In [2]:
from dataGadgets import *

import numpy as np
import matplotlib.pyplot as plt
import scipy.optimize as opt
import pandas as pd
Functions successfully imported!



In [3]:
# unit: g/mL at 25 °C
ρ_h20 = 1
ρ_tol = 0.8623
ρ_eth = 0.7849

def V_water_toluene_ethanol(V_h2o,V_tol,V_eth):
    global ρ_h20,ρ_tol,ρ_eth
    m_h20 = V_h2o*ρ_h20
    m_tol = V_tol*ρ_tol
    m_eth = V_eth*ρ_eth
    mTotal = m_h20+m_tol+m_eth
    
    frac_h20 = m_h20/mTotal
    frac_tol=m_tol/mTotal
    frac_eth=m_eth/mTotal
    
    frac_h20 = "{0:.3g}".format(frac_h20)
    frac_tol="{0:.3g}".format(frac_tol)
    frac_eth="{0:.3g}".format(frac_eth)
    return print(
                'Table format for mass fraction:','|', frac_eth, '|',frac_h20, '|',frac_tol,'|\n',\
                'Table format for component volume:','|', V_h2o, '|',V_tol, '|',V_eth,'|'
                )

In [8]:
# type in the volume in ml for the three components
# Copy the string to the cells for the tables below.
V_water_toluene_ethanol(16.5,1,16.7)

Table format for mass fraction: | 0.43 | 0.542 | 0.0283 |
 Table format for component volume: | 16.5 | 1 | 16.7 |


__Double click to edit the tables__
<p>Ensure correspondence between upper and lower data in the table!!</p>

| V_water | V_toluene | V_ethanol |
|:-:|:-:|:-:|
| 4 | 25.4 | 22.1 |
| 10 | 11.5 | 32.8 |
| 10 | 4.9 | 24.6 |
| 16.5 | 1 | 16.7 |

| frac_ethanol | frac_water | frac_toluene |
|:-:|:-:|:-:|
| 0.401 | 0.0925 | 0.506 |
| 0.564 | 0.219 | 0.217 |
| 0.576 | 0.298 | 0.126 |
| 0.43 | 0.542 | 0.0283 |